### PR TITLE
DOC: update PeriodArray docstring

### LINKED ISSUE #37237: DOC: update PeriodArray docstring
AFAICT we currently don't have dedicated pages for these methods in the API reference. 

When we add them we also want add links on the `pd.PeriodArray` [page](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.arrays.PeriodArray.html) (and possibly in other places)

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #37237 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/period.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ class PeriodArray(PeriodMixin, dtl.DatelikeOps):
     """
     Pandas ExtensionArray for storing Period data.
 
-    Users should use :func:`period_array` to create new instances.
+    Users should use :func:`period_range` to create new instances.
```
Comment (jbrockmendel) [inline pandas/core/arrays/period.py:70]: period_array is correct

Comment (jorisvandenbossche) [reply pandas/core/arrays/period.py:70]: It's not public, see also the linked issue

**Review by jorisvandenbossche — COMMENTED** [review]
@arw2019 thanks for the PR. 
I would also mention `pd.array(..)` as the way to create a PeriodArray from a sequence of Period scalars.

**Comment by arw2019** [comment]
> @arw2019 thanks for the PR.
> I would also mention `pd.array(..)` as the way to create a PeriodArray from a sequence of Period scalars.

Added

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks for the update!

**Comment by jorisvandenbossche** [comment]
Small development related comment: can you generally try to avoid rebasing / force pushing to a branch, but rather only add new commits? (which means merging master instead of rebasing to update to latest master) Github works better that way (eg you can specifically see the new changes since your last review) making reviewing easier. 
(not of any concern of course for this specific PR of small size ;))

**Pull request merged**